python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -234,53 +245,72 @@ def _add_bundled_input(self, refs: RefBundle):
         """
         raise NotImplementedError
 
-    def _handle_task_submitted(self, task: "_TaskState"):
-        """Handle a newly submitted task, notifying the output queue and updating
-        object store metrics.
-
-        This should be called by subclasses right after a task is submitted.
-
-        Args:
-            task: The task state for the newly submitted task.
-        """
-        # Notify output queue that this task is pending.
-        self._output_queue.notify_pending_task(task)
-
-    @abstractmethod
-    def notify_work_completed(
-        self, ref: Union[ObjectRef[ObjectRefGenerator], ray.ObjectRef]
+    def _submit_data_task(
+        self,
+        gen: StreamingObjectRefGenerator,
+        inputs: RefBundle,
+        task_done_callback: Optional[Callable[[], None]] = None,
     ):
-        """Indicates that a task is done executing OR that a worker is done starting.
-
-        This must be implemented by subclasses.
+        """Submit a new data-handling task."""
+        # TODO(hchen):
+        # 1. Move this to the base PhyscialOperator class.
+        # 2. This method should only take a block-processing function as input,
+        #    instead of a streaming generator. The logic of submitting ray tasks
+        #    can also be capsulated in the base class.
```

Comment:
yes this makes more sense to me, right now `_submit_data_task` is just doing bookkeeping here, the task already being submitted when creating `gen`.